java/test/org/openqa/selenium/support/decorators/DecoratedWebDriverTest.java

```diff
@@ -163,6 +164,45 @@ void findElement() {
     verifyDecoratingFunction($ -> $.findElement(By.id("test")), found, WebElement::click);
   }
 
+  @Test
+  void doesNotCreateTooManyClasses() {
+    final WebElement found = mock(WebElement.class);
+    Function<WebDriver, WebElement> f = $ -> $.findElement(By.id("test"));
+    Fixture fixture = new Fixture();
+    when(f.apply(fixture.original)).thenReturn(found);
+
+    WebElement proxy1 = f.apply(fixture.decorated);
```

Comment:
should we also verify that decorated classes for different WebElement instances are still the same?

Reply:
thats also a good improvement, i have added this to the tests.